.vib/clickhouse/ginkgo/clickhouse_suite_test.go

```diff
@@ -0,0 +1,71 @@
+package clickhouse_test
+
+import (
+	"context"
+	"flag"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	batchv1 "k8s.io/api/batch/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+var (
+	kubeconfig     string
+	releaseName    string
+	namespace      string
+	username       string
+	password       string
+	shards         int
+	timeoutSeconds int
+	timeout        time.Duration
+)
+
+func init() {
+	flag.StringVar(&kubeconfig, "kubeconfig", "", "absolute path to the kubeconfig file")
+	flag.StringVar(&releaseName, "name", "", "name of the primary statefulset")
+	flag.StringVar(&namespace, "namespace", "", "namespace where the application is running")
+	flag.StringVar(&username, "username", "", "database user")
+	flag.StringVar(&password, "password", "", "database password for username")
+	flag.IntVar(&shards, "shards", 2, "number of shards")
+	flag.IntVar(&timeoutSeconds, "timeout", 120, "timeout in seconds")
+	timeout = time.Duration(timeoutSeconds) * time.Second
+}
+
+func TestClickHouse(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "ClickHouse Persistence Test Suite")
+}
+
+func createJob(ctx context.Context, c kubernetes.Interface, name, port, image, stmt string) error {
+	job := &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		TypeMeta: metav1.TypeMeta{
+			Kind: "Job",
+		},
+		Spec: batchv1.JobSpec{
+			Template: v1.PodTemplateSpec{
+				Spec: v1.PodSpec{
+					RestartPolicy: "Never",
+					Containers: []v1.Container{
+						{
+							Name:    "clickhouse",
+							Image:   image,
+							Command: []string{"clickhouse-client", "--user", username, "--password", password, "--multiquery", "--host", releaseName, "--port", port, "--query", stmt},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	_, err := c.BatchV1().Jobs(namespace).Create(ctx, job, metav1.CreateOptions{})
+
+	return err
+}
```

Comment:
Shouldn't this be a generic function (k8s.go) that receives the PodSpec as argument?